[Secuity Solution][DQD] add list view and latest flyout (Phase 1)

kapral18 opened the pull request:
addresses #185881

## Data Quality Dashboard UI Overhaul (Phase 1)

This PR introduces UI changes to the Data Quality Dashboard, focusing on improving user experience and improving on existing data quality check process.

## Notable changes:

- Update List View UI
- Move in-row check expansion into a Flyout
- Remove summary tab from index check 
- Update index check UI
- Add table action to trigger manual individual index check without opening  index check properties
- Add in-flyout button to trigger manual individual check
- Add additional index stats panel within flyout

## Notable technical changes:

- remove prop drilling of new and existing omnipresent props by unifying them in a series of context providers at the top of DQ dashboard
- introduce TestDataQualityProviders separate from external TestProviders (renamed to TestExternalProviders) in tests. Change affected tests.
- introduce `useIndicesCheck` hook to unify logic of index checking to be able to call index checking from anywhere within DQD code without relying on flaky and complicated useEffect driven logic of state updates.
- introduce `useIsMounted`, hook to resolve issues with react state update leaks
- introduce throttled `useCurrentWindowWidth` to handle custom index flyout sizing for different screens.
- cleanup and refactor whatever is directly related or affected by aforementioned code/test changes (including traces of removal of summary tab)
- add extensive behavioral unit tests

# UI Changes (Before / After)

## List View Layout made more spacious
![image](https://github.com/user-attachments/assets/8cc25285-9a7b-45d9-82f2-55bb7ad4680b)

## Check All Layout made more spacious
![image](https://github.com/user-attachments/assets/c254cfa9-92cf-4971-9737-9c85e253696e)

## Total Stats Rollup converted to badges
![image](https://github.com/user-attachments/assets/68831c84-274d-40d8-b9fa-813e169e8f1e)

## ILM Phase Filter repositioned
![image](https://github.com/user-attachments/assets/85227737-1217-48b9-82b9-dc2d6a24e58c)

## Pattern Title Section

- Rearranged into accordion trigger button. Initially open by default.
- IlmPhase badges and Pattern title are now  stacked horizontally.
- Index check result emoji converted to 'Pass' | 'Fail' badge
- Pattern stats text rearranged as badged text. Incompatible fields show as red badges (when present), the rest - hollow.

![image](https://github.com/user-attachments/assets/ca3c6e3e-a7ed-4a49-a7a6-9ea4842ac2de)

## Latest Pattern Indices Check Table

- Added a new actions column with 2 actions (from left to right): 
    - View details (replaces row expander functionality (and icon) and instead opens the index check results in a flyout)
    - Check now **(NEW)** (adds ability to inline check the index without opening it.)
- Index check result emoji turned to 'Pass' | 'Fail' badge
- `IlmPhase`, `Size`, `Last Check` columns width is shrunk to give more space for index name

![image](https://github.com/user-attachments/assets/d9c83d9b-de3a-4153-9ed7-a9823e62f67d)

## Flyout Header

- Added index name with result check badge as title
- Added last check time as subtitle
- Added Tabs section for Latest Check and History **(REMOVED in latest revision)**

![image](https://github.com/user-attachments/assets/b66617de-2b59-44d5-908b-3fabb3f7087e)

## Flyout Stats Panel

- Added new index stats panel

![image](https://github.com/user-attachments/assets/f21974f5-72b5-4a38-90de-f2ff5b9d1fd0)

## Index Check Fields Tab

- Tabs converted to a button group
- Summary Tab is **REMOVED**
- All field count badges have hollow color, except for red color for  `incompatible fields`  tab (when count > 0) and `ecs compliant fields` tab (when `@timestamp` is missing)

![image](https://github.com/user-attachments/assets/e78218d4-237f-4ad1-95f3-0eb4d57356fa)

## Index Check Fields Callouts

- Callout header is removed (to avoid duplication with active tab name)
- Actions are converted into sticky footer (shows when scrolled sticky to bottom, otherwise renders after the table)
- Same for every index check fields tab

![image](https://github.com/user-attachments/assets/d8a192e8-cdfd-4691-b862-91abdedcb9b8)

## Compare Table List Values

- List values in compare tables are now horizontally stacked instead of vertical to save space (applies to all compare tables in each index check fields tab)

![image](https://github.com/user-attachments/assets/7ef3341d-1be9-41e7-a789-15262fac6de6)

## Compare Table Columns

- `ECS description` field width increased at the expense of `field` field,  to make room for more readable description (applies respectively to all compare tables within index check fields tabs)

![image](https://github.com/user-attachments/assets/b45235a9-cf3d-4b6d-b65a-eaadc577cf2f)

## Flyout footer

- Add `Check now` button, that checks currently open index again and updates the results in place.

![image](https://github.com/user-attachments/assets/02b9a63d-6c0c-4eff-aeb6-452ae78f28ea)

kapral18 committed:
[Security Solution][DQD] add list view and latest flyout (Phase 1)

elasticmachine commented:
Pinging @elastic/security-threat-hunting-explore (Team:Threat Hunting:Explore)

elasticmachine commented:
Pinging @elastic/security-threat-hunting (Team:Threat Hunting)

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/221861)
* Commit: a324db8c4aa79f9bbe466c3a4457b3b688675e94

### Failed CI Steps
* [Quick Checks](https://buildkite.com/elastic/kibana-pull-request/builds/221861#0190bca9-fe6c-4473-832d-75d6dac1f3ea)

cc @kapral18

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/221861","number":221861,"commit":"a324db8c4aa79f9bbe466c3a4457b3b688675e94"}],"number":221861}
buildkite-pr-comment-->

agusruidiazgd commented on x-pack/packages/security-solution/ecs_data_quality_dashboard/impl/data_quality/data_quality_panel/data_quality_summary/ilm_phase_filter/styles.tsx:
```diff
@@ -0,0 +1,28 @@
+  width: 100%;
+`;
+
+export const StyledOptionLabel = styled.span`
+  font-weight: bold;
+`;
+
+export const StyledFormControlLayout = styled(EuiFormControlLayout)`
+  height: 42px;
```
Could we use the theme value for this styles? something like `theme.eui.euiFontWeightBold` and `theme.eui.euiSize`

kapral18 replied:
fixed, thanks

agusruidiazgd commented on x-pack/packages/security-solution/ecs_data_quality_dashboard/impl/data_quality/data_quality_panel/data_quality_summary/ilm_phase_filter/styles.tsx:
```diff
@@ -0,0 +1,28 @@
+`;
+
+export const StyledFormControlLayout = styled(EuiFormControlLayout)`
+  height: 42px;
```
Here I think the `theme.eui.euiSize...` could be used

kapral18 replied:
Tbh I don't have the context of this change, since I didn't do it, this came over from the old code. I assume this was an attempt at fixing the height of EuiComboBox, so that it doesn't swallow the border below. So I would abstain from making it dynamic, because it might not match the measure once changed. Wdyt?

agusruidiazgd commented on x-pack/packages/security-solution/ecs_data_quality_dashboard/impl/data_quality/data_quality_panel/pattern/index_check_flyout/index.tsx:
```diff
@@ -0,0 +1,171 @@
+    };
+  }, []);
+
+  const tabs = [
+    {
+      id: LATEST_CHECK,
+      name: LATEST_CHECK,
+    },
+    {
+      id: HISTORY,
+      name: HISTORY,
+      disabled: true,
+    },
+  ];
+
+  const renderTabs = tabs.map((tab, index) => (
+    <EuiTab
+      onClick={() => setSelectedTabId(tab.id)}
+      isSelected={tab.id === selectedTabId}
+      key={index}
+      disabled={tab.disabled}
+    >
+      {tab.name}
+    </EuiTab>
+  ));
```
What do you think about using `useMemo` for `tabs` and `useCallback` for `renderTabs` ? 
So in this way helps with performance by ensuring that tabs and renderTabs are only recalculated when necessary, reducing unnecessary re-renders

kapral18 replied:
Hey @agusruidiazgd thanks for your suggestion.

What do you think about:

1. Extracting `tabs` out of component, because it has no dependencies and has no reason to be in the render cycle.
2. Use `useMemo()` for `renderTabs` to cache it. `useCallback` here will not help, because identity of mapCallback has no affect on return value, you will still do the `map`ping part over and over again, if we talk about performance, so we need to memoize the return value not the function itself.

Let me know wdyt about this.

agusruidiazgd replied:
perfect!

agusruidiazgd reviewed (commented):
Amazing work Karen! 💫 I left some minor comments

angorayc reviewed (commented):
Looking great so far, will continue the review and do more testing locally.

angorayc commented:
@kapral18 Some issues found on the serverless version with `viewer` role, could you please check if you can reproduce them, thank you!

<img width="2560" alt="Screenshot 2024-08-15 at 10 25 13-serverless" src="https://github.com/user-attachments/assets/fc28d198-5c96-45af-9370-d3b3f23e0edf">

kapral18 commented:
> @kapral18 Some issues found on the serverless version with `viewer` role, could you please check if you can reproduce them, thank you!
> 
> <img alt="Screenshot 2024-08-15 at 10 25 13-serverless" width="2000" src="https://private-user-images.githubusercontent.com/6295984/358182260-fc28d198-5c96-45af-9370-d3b3f23e0edf.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjM3Mjc5NDksIm5iZiI6MTcyMzcyNzY0OSwicGF0aCI6Ii82Mjk1OTg0LzM1ODE4MjI2MC1mYzI4ZDE5OC01Yzk2LTQ1YWYtOTM3MC1kM2IzZjIzZTBlZGYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgxNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MTVUMTMxNDA5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YzNiYmMyYTQyNGM1Y2E2MGM3YTRiMWQyNDE2ZDAxZjdkYTgwY2VjYTBhZDQ5YjBiYzA1ZmU1OWI0NzEwZTg4NiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.qpdjqF8oG4tx9HGj3q4IUQv2ezIdieDYaZeubCijm4M">

Hey @angorayc thanks for the comment, I can reproduce this.

This behavior has existed in original code, so this is not caused by the changes from this PR and is not related to this ticket.

Here is a screenshot from serverless project running on main branch
![image](https://github.com/user-attachments/assets/badddb5a-565c-4afc-b959-439a664ee0e2)
![image](https://github.com/user-attachments/assets/08882ff1-7b9f-4941-8adb-217368884148)

andrew-goldstein commented on x-pack/packages/security-solution/ecs_data_quality_dashboard/impl/data_quality/data_quality_panel/index_properties/index_check_fields/index.tsx:
```diff
@@ -0,0 +1,120 @@
+`;
+
+const StyledButtonGroup = styled(EuiButtonGroup)`
+  button[data-test-subj='incompatibleTab'] {
```
nit: consider alternatives to referencing `data-test-subj`s in styles

kapral18 replied:
why? and what alternative do you see here?

andrew-goldstein replied:
`data-test-subj` is a reliable selector, but it's _uncommon_ to see it used as a selector in _production_ code in Kibana.

How uncommon?

The following `grep` finds code (in `$KIBANA_HOME`) that uses `data-test-subj` as a `button` selector:

```sh
➜  kibana git:(attack_discovery_output_token_limits) ✗ grep -r --include="*.test.tsx" 'button\[data-test-subj=' .
./packages/home/sample_data_card/src/footer/remove_footer.test.tsx:      component.find(`button[data-test-subj="removeSampleDataSet${id}"]`).simulate('click');
./packages/home/sample_data_card/src/footer/remove_footer.test.tsx:      component.find(`button[data-test-subj="removeSampleDataSet${id}"]`).simulate('click');
./packages/home/sample_data_card/src/footer/install_footer.test.tsx:      component.find(`button[data-test-subj="addSampleDataSet${id}"]`).simulate('click');
./packages/home/sample_data_card/src/footer/install_footer.test.tsx:      component.find(`button[data-test-subj="addSampleDataSet${id}"]`).simulate('click');
# ... output truncated for brevity
```

when the output above is piped to `wc`, we can count the usages to see if it's common.

There are `329` usages In just _test_ files `*.test.tsx` :

```sh
➜  kibana git:(feat/DQD/185881-add-list-view-and-latest-flyout) ✗ grep -r --include="*.test.tsx" 'button\[data-test-subj=' . | wc -l
     329
```

There are `331` usages when we also include _production_ files `*.tsx` :

```sh
➜  kibana git:(feat/DQD/185881-add-list-view-and-latest-flyout) ✗ grep -r --include="*.tsx" 'button\[data-test-subj=' . | wc -l
     331
```

Where do the additional two production usages come from?

```sh
➜  kibana git:(feat/DQD/185881-add-list-view-and-latest-flyout) ✗ diff <(grep -r --include="*.test.tsx" 'button\[data-test-subj=' .) <(grep -r --include="*.tsx" 'button\[data-test-subj=' .)
316a317,318
> ./x-pack/packages/security-solution/ecs_data_quality_dashboard/impl/data_quality/data_quality_panel/index_properties/index_check_fields/index.tsx:  button[data-test-subj='incompatibleTab'] {
> ./x-pack/packages/security-solution/ecs_data_quality_dashboard/impl/data_quality/data_quality_panel/index_properties/index_check_fields/index.tsx:  button[data-test-subj='ecsCompliantTab'] {
```

Per the `diff` output above, the only two _production_ usages found are the changes proposed in this PR.

It's ok if there's no better alternative, as this may be the first time it's _required_ (as a button selector) in Kibana's production code.

kapral18 replied:
I would argue if we are assessing the usage of `data-test-subj` in styles better grep that specifically like

`grep -r --include="*.tsx" --exclude="*.test.ts*" '\[data-test-subj.*\] {' -A1`

![image](https://github.com/user-attachments/assets/a25d2ec6-e03e-4da3-9446-6720f003d102)

As seen in the screenshot there is more usage than introduced in this PR. 

Of course it's not common, but I am not sure commonality is a strong factor, even tho it's a factor. I would say as a general rule as long as it makes sense and doesn't bring more problems than it solves including stylistic ones, why not? Wdyt?

In this specific case I don't see a more elegant hook into button tab within button group. But feel free to share if you see.

kapral18 commented on x-pack/packages/security-solution/ecs_data_quality_dashboard/impl/data_quality/data_quality_panel/data_quality_summary/index.tsx:
```diff
@@ -6,108 +6,75 @@
+          <StyledRollupContainer>
+            <StatsRollup
+              docsCount={totalDocsCount}
+              incompatible={totalIncompatible}
+              indices={totalIndices}
+              indicesChecked={totalIndicesChecked}
+              sizeInBytes={totalSizeInBytes}
```
yes, because StatsRollup is a shared component and it receives differents sets of those props:

- Here it receives the total rollup stats
- In PatternSummary it receives pattern level rollup stats

kapral18 commented on x-pack/packages/security-solution/ecs_data_quality_dashboard/impl/data_quality/data_quality_panel/body/data_quality_details/indices_details/index.tsx:
```diff
@@ -5,118 +5,51 @@
+
+  return (
+    <div data-test-subj="indicesDetails">
+      {patterns.map((pattern, i) => (
```
@angorayc it was a leftover from original code, I didn't notice. Will fix thanks.

kapral18 commented on x-pack/packages/security-solution/ecs_data_quality_dashboard/impl/data_quality/data_quality_panel/data_quality_summary/index.tsx:
```diff
@@ -6,108 +6,75 @@
-}
+const StyledIlmPhaseFilterContainer = styled.div`
+  width: 100%;
+  max-width: 432px;
```
I would actually put those constants back into styles :D Because there is no reason for them to be outside besides enforcing the type, which to me is excessive here. Namewise it's clear what those variables are because the css rule explicitly states that those are `min-width` and `max-width`  and furthermore enclosing styledcomponent and name tells explicitly where they belong `SummaryActionsContainerFlexItem`

So to me this feels a little like:

```tsx
// this is a variable
const variable = 'variable'
```

:D

But good question 

Wdyt? 

cc: @angorayc

andrew-goldstein commented on x-pack/packages/security-solution/ecs_data_quality_dashboard/impl/data_quality/data_quality_panel/index_result_badge/index.tsx:
```diff
@@ -0,0 +1,49 @@
+        color={getIndexResultBadgeColor(incompatible)}
+        {...props}
+      >
+        {incompatible > 0 ? 'Fail' : 'Pass'}
```
Consider returning an `i18n` value here

kapral18 replied:
yessss good catch, thanks

andrew-goldstein commented on x-pack/packages/security-solution/ecs_data_quality_dashboard/impl/data_quality/mock/test_providers/test_providers.tsx:
```diff
@@ -75,21 +88,221 @@ export const TestProvidersComponent: React.FC<Props> = ({ children, isILMAvailab
+    checkState: {},
+  };
+
+  for (const key of Object.keys(mergedResultsRollupContextProps.patternIndexNames)) {
+    for (const indexName of mergedResultsRollupContextProps.patternIndexNames[key]) {
+      const mappingsProperties = getMappingsProperties({
+        indexName,
+        indexes: mockMappingsResponse as Record<string, IndicesGetMappingIndexMappingRecord>,
+      });
+      const unallowedValues = getUnallowedValues({
+        requestItems: getUnallowedValueRequestItems({
+          ecsMetadata: EcsFlatTyped,
+          indexName,
+        }),
+        searchResults: mockUnallowedValuesResponse as unknown as UnallowedValueSearchResult[],
+      });
+      const partitionedFieldMetadata = getSortedPartitionedFieldMetadata({
+        ecsMetadata: EcsFlatTyped,
+        loadingMappings: false,
+        mappingsProperties,
+        unallowedValues,
+      });
+      mergedIndicesCheckContextProps.checkState[indexName] = {
+        isChecking: false,
+        isLoadingMappings: false,
+        isLoadingUnallowedValues: false,
+        indexes: mockMappingsResponse as Record<string, IndicesGetMappingIndexMappingRecord>,
+        partitionedFieldMetadata,
+        searchResults: mockUnallowedValuesResponse as unknown as UnallowedValueSearchResult[],
+        unallowedValues,
+        mappingsProperties,
+        genericError: null,
+        mappingsError: null,
+        unallowedValuesError: null,
+        isCheckComplete: true,
+      };
+    }
+  }
+
+  Object.assign(mergedIndicesCheckContextProps, indicesCheckContextProps);
```
nit: Consider eliminating local mutation of `mergedIndicesCheckContextProps.state` and mutation via `Object.assign` with something like:

```typescript
  const mergedIndicesCheckContextProps: UseIndicesCheckReturnValue = {
    checkIndex: jest.fn(),
    checkState: Object.keys(
      mergedResultsRollupContextProps.patternIndexNames
    ).reduce<UseIndicesCheckCheckState>((acc, key) => {
      mergedResultsRollupContextProps.patternIndexNames[key].forEach((indexName) => {
        const mappingsProperties = getMappingsProperties({
          indexName,
          indexes: mockMappingsResponse as Record<string, IndicesGetMappingIndexMappingRecord>,
        });

        const unallowedValues = getUnallowedValues({
          requestItems: getUnallowedValueRequestItems({
            ecsMetadata: EcsFlatTyped,
            indexName,
          }),
          searchResults: mockUnallowedValuesResponse as unknown as UnallowedValueSearchResult[],
        });

        const partitionedFieldMetadata = getSortedPartitionedFieldMetadata({
          ecsMetadata: EcsFlatTyped,
          loadingMappings: false,
          mappingsProperties,
          unallowedValues,
        });

        acc[indexName] = {
          isChecking: false,
          isLoadingMappings: false,
          isLoadingUnallowedValues: false,
          indexes: mockMappingsResponse as Record<string, IndicesGetMappingIndexMappingRecord>,
          partitionedFieldMetadata,
          searchResults: mockUnallowedValuesResponse as unknown as UnallowedValueSearchResult[],
          unallowedValues,
          mappingsProperties,
          genericError: null,
          mappingsError: null,
          unallowedValuesError: null,
          isCheckComplete: true,
        };
      });

      return acc;
    }, {}),
    ...indicesCheckContextProps,
  };
```

kapral18 replied:
actually I decided to build on this and fully refactor this gigantic function let me know what you think. I pushed the changes, and thanks for looking through this.

kapral18 committed:
chore(comments): address comments

- refactor index check flyout tabs rendering
- simplify test data quality providers setup
- index result badge to use i18n translations

andrew-goldstein reviewed (approved):
Thanks @kapral18  for these improvements! 🙏 
✅ Desk tested locally
LGTM 🚀

andrew-goldstein commented on x-pack/packages/security-solution/ecs_data_quality_dashboard/impl/data_quality/data_quality_panel/pattern/index_check_flyout/index.tsx:
```diff
@@ -0,0 +1,175 @@
+  {
+    id: HISTORY,
+    name: HISTORY,
+    disabled: true,
```
Consider hiding the `History` tab in the screenshot below, because it's "coming soon", but not implemented:

![history](https://github.com/user-attachments/assets/4a6c2bb9-474d-4528-ba5e-174ee67e906e)

kapral18 replied:
if I hide it will have to hide Latest Check as well, because tabline with a single tab is weird. That's why I presented this in last DQD meeting as is and checked with Dhru whether this looks ok. From what I remember this was approved. but maybe @dhru42 can clarify one more time?

dhru42 replied:
We shouldn't show anything we're not releasing in the UI. Main reason is anything greyed out means that the user doesn't have the right subscription to access it. So for users who are on the highest subscription (Enterprise), that shows they dont have access to it which is a point of confusion.

dhru42 replied:
Since this historical functionality + redesign is releasing publically in 8.16 (together), is there a reason why we're having this discussion?

@andrew-goldstein @kapral18 maybe I'm missing context here.

kapral18 replied:
@dhru42 thanks for the context, I didn't know about the greyed out == lack of subscription.

Why we are having this discussion is because once I merge this it will appear in serverless release a week later and is generally ready for deploy. And this is not under a feature flag because this is not a new functionality but rather a UI overhaul which is unfeasable to hide behind a flag.

Ok so should I just remove the history tab or remove the entire tabline?

kapral18 replied:
@dhru42 @andrew-goldstein removed the tabline it looks like this now
![image](https://github.com/user-attachments/assets/3e27e86b-09b9-428d-8a32-8a2172d1ad17)

angorayc reviewed (approved):
As issues found during testing are not introduced by this PR, let's follow them up later. https://github.com/elastic/kibana/issues/164832

Thanks for implementing the changes, looking great!

angorayc commented:
> @kapral18 Some issues found on the serverless version with `viewer` role, could you please check if you can reproduce them, thank you!
> 
> <img alt="Screenshot 2024-08-15 at 10 25 13-serverless" width="2000" src="https://private-user-images.githubusercontent.com/6295984/358182260-fc28d198-5c96-45af-9370-d3b3f23e0edf.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQwNTc0MDcsIm5iZiI6MTcyNDA1NzEwNywicGF0aCI6Ii82Mjk1OTg0LzM1ODE4MjI2MC1mYzI4ZDE5OC01Yzk2LTQ1YWYtOTM3MC1kM2IzZjIzZTBlZGYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgxOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MTlUMDg0NTA3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NDQxY2RjZWYxYWI5NzM5ZDE2NDYwMzhjZjc4Mzc3MDc5YjIyYzVhN2RjN2Y4YzhkODQ2NmFhYjg0YzkwNDVkNCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.Wto2fBfpdOOqWgjBTX0KZsuqK94bExmtXGTltI8rCQU">

Clarification on `Viewer should not have access to alerts index`: When I saw this screen, I thought if a viewer has no right to see the historical result of auditbeat-* , then it should have no right to see the historical result of alerts index. But having another look, it makes sense as the index previlege does varies by index. A user can have read access to alert index but have no read access to auditbeat

kapral18 committed:
chore(comments): fix padding issue mentioned in comments

kapral18 commented:
> @kapral18 Some issues found on the serverless version with `viewer` role, could you please check if you can reproduce them, thank you!
> 
> <img alt="Screenshot 2024-08-15 at 10 25 13-serverless" width="2000" src="https://private-user-images.githubusercontent.com/6295984/358182260-fc28d198-5c96-45af-9370-d3b3f23e0edf.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQwNTc1MjUsIm5iZiI6MTcyNDA1NzIyNSwicGF0aCI6Ii82Mjk1OTg0LzM1ODE4MjI2MC1mYzI4ZDE5OC01Yzk2LTQ1YWYtOTM3MC1kM2IzZjIzZTBlZGYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgxOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MTlUMDg0NzA1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NTM1NWNmZTllNjBjMTlhYTJlMWRkMzA3OTFlNzc0ZjRmYTUwMzdkMjU4MzM2ZTk1ZjJiNzY2NWZlOTg0NGIzMCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.wc8OmlTg_PNiBpZ9bgbTC0K1pqQgB1c0LtpZT-G4ZE4">

The padding issue is addressed in a new separate [commit](https://github.com/elastic/kibana/pull/188468/commits/632e91525342ae78ded9725b4782f2f0f6bafbad). Thanks for checking.

kapral18 commented:
> @kapral18 Some issues found on the serverless version with `viewer` role, could you please check if you can reproduce them, thank you!
> 
> <img alt="Screenshot 2024-08-15 at 10 25 13-serverless" width="2000" src="https://private-user-images.githubusercontent.com/6295984/358182260-fc28d198-5c96-45af-9370-d3b3f23e0edf.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQwNjUxODgsIm5iZiI6MTcyNDA2NDg4OCwicGF0aCI6Ii82Mjk1OTg0LzM1ODE4MjI2MC1mYzI4ZDE5OC01Yzk2LTQ1YWYtOTM3MC1kM2IzZjIzZTBlZGYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgxOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MTlUMTA1NDQ4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZTUxNGYzM2UwN2Y0ZjY4NjBmOTUyYzc0NmZhNDg1MjgyY2M0MDMzNTk4ODAzZDAyNmUyYTMwNTY0YTAzN2NlZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.4kvYY7EW7Fj4Tdh_HuFwNP0FyGrOWTyPGtQt3aemxdo">

Added this https://github.com/elastic/kibana/issues/190687 to fix the incorrect results label issue

kapral18 committed:
improve(storage-treemap): add padding to storage-treemap no data view

kapral18 commented:
> @kapral18 Some issues found on the serverless version with `viewer` role, could you please check if you can reproduce them, thank you!
> 
> <img alt="Screenshot 2024-08-15 at 10 25 13-serverless" width="2000" src="https://private-user-images.githubusercontent.com/6295984/358182260-fc28d198-5c96-45af-9370-d3b3f23e0edf.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQwNjYyNjUsIm5iZiI6MTcyNDA2NTk2NSwicGF0aCI6Ii82Mjk1OTg0LzM1ODE4MjI2MC1mYzI4ZDE5OC01Yzk2LTQ1YWYtOTM3MC1kM2IzZjIzZTBlZGYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgxOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MTlUMTExMjQ1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NWRlMzk5ZjJlNmEyMzRmZjBmNWZkMWQ0MDc2MDkwYmY4ZWE0MTcwZGNmMTE2YWIwMGMxYjQ3ODE5NzRiMGIyNCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.Cs93Rbyvrh-BNHMgnJTf0PdcCtU5wIuSlVbeP00GukI">

@angorayc I also noticed that the no data label when storage-treemap is missing is being smushed in the design because of outlined borders of pattern accordions, so I added some paddings per this commit https://github.com/elastic/kibana/pull/188468/commits/906d3ef792be5eca6549dd0fb49f82a6d0cc511f

now it looks like this
![image](https://github.com/user-attachments/assets/7a77cda0-ae14-4470-8826-66783584174a)

kapral18 committed:
refactor: remove tab rendering and related tests from IndexCheckFlyoutComponent

kapral18 committed:
feat: refactor index check flyout translations and tooltips

- Moved CHECK_NOW translation to a new file for better organization.
- Updated tooltip translations for various stats in the StatsRollup
component.
- Improved consistency in tooltip messages across the application.
- Renamed actions in the summary table from "Check now" to "Check index"
and "View details" to "View check details"

kapral18 commented:
hey @angorayc @agusruidiazgd @andrew-goldstein 

We just revisited the tooltip texts with docs people
so one last commit https://github.com/elastic/kibana/pull/188468/commits/99866bb7339a83860c8587ffa89817b25946afad

kapral18 commented:
docs PR is ready https://github.com/elastic/security-docs/pull/5718
